paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/CommentTrackingXmlPullParser.kt

```diff
@@ -0,0 +1,149 @@
+ *
+ * An [XmlPullParser] that keeps track of the last comment preceding an XML tag and special comments
+ * that are used in the framework resource files for describing groups of "attr" resources. Here is
+ * an example of an "attr" group comment:
+ * <pre>
+ * &lt;!-- =========== --&gt;
+ * &lt;!-- Text styles --&gt;
+ * &lt;!-- =========== --&gt;
+ * &lt;eat-comment/&gt;
+</pre> *
+ */
+open class CommentTrackingXmlPullParser : KXmlParser() {
+  /**
+   * Returns the last encountered comment that is not an ASCII art.
+   */
+  var lastComment: String? = null
+  private var tagEncounteredAfterComment: Boolean = false
+  private val attrGroupCommentStack = ArrayList<String?>(4)
+
+  /**
+   * Initializes the parser. XML namespaces are supported by default.
+   */
+  init {
+    try {
+      setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, true)
+    } catch (e: XmlPullParserException) {
+      throw Error(e) // KXmlParser is guaranteed to support FEATURE_PROCESS_NAMESPACES.
+    }
+  }
+
+  /**
+   * Returns the name of the current "attr" group, e.g. "Button Styles" group for "buttonStyleSmall" "attr" tag.
+   */
+  fun getAttrGroupComment(): String? = attrGroupCommentStack[attrGroupCommentStack.size - 1]
+
+  @Throws(XmlPullParserException::class, IOException::class)
+  override fun nextToken(): Int {
+    val token = super.nextToken()
+    processToken(token)
+    return token
+  }
+
+  @Throws(XmlPullParserException::class, IOException::class)
+  override operator fun next(): Int {
+    throw UnsupportedOperationException("Use nextToken() instead of next() for comment tracking to work")
+  }
+
+  private fun processToken(token: Int) {
+    when (token) {
+      XmlPullParser.START_TAG -> {
+        if (tagEncounteredAfterComment) {
+          lastComment = null
+        }
+        tagEncounteredAfterComment = true
+        // Duplicate the last element in attrGroupCommentStack.
+        attrGroupCommentStack.add(attrGroupCommentStack[attrGroupCommentStack.size - 1])
+        assert(attrGroupCommentStack.size == depth + 1)
+        if (TAG_EAT_COMMENT == name && prefix == null) {
+          // The framework attribute file follows a special convention where related attributes are grouped together,
+          // and there is always a set of comments that indicate these sections which look like this:
+          //     <!-- =========== -->
+          //     <!-- Text styles -->
+          //     <!-- =========== -->
+          //     <eat-comment/>
+          // These section headers are always immediately followed by an <eat-comment>. Not all <eat-comment/> sections are
+          // actually attribute headers, some are comments. We identify these by looking at the line length; category comments
+          // are short, and descriptive comments are longer.
+          if (lastComment != null && lastComment!!.length <= ATTR_GROUP_MAX_CHARACTERS && !lastComment!!.startsWith(
+              "TODO:"
+            )
+          ) {
```

Comment:
```suggestion
          if (lastComment != null && lastComment!!.length <= ATTR_GROUP_MAX_CHARACTERS && !lastComment!!.startsWith("TODO:")) {
```

or basically, any way that the line breaks can be setup without "TODO:" on its own line?